include/LightGBM/objective_function.h

```diff
@@ -9,6 +9,11 @@
 #include <LightGBM/dataset.h>
 #include <LightGBM/meta.h>
 #include <LightGBM/utils/constrained.hpp>
+#include <LightGBM/proxy-loss-types/proxy_loss_base.hpp>
```

Comment:
1. Do not use "-" in folder names, replace with "_"'s.
2. We should not be adding more stuff to objective_function.h but instead to constrained_objective_function.h? Or this becomes unmaintainable when we try to merge newer patches from LGBM